scour/scour.py

```diff
@@ -3627,8 +3625,8 @@ def scourString(in_string, options=None):
     # on the first pass, so we do it multiple times
     # does it have to do with removal of children affecting the childlist?
     if options.keep_editor_data is False:
-        while removeNamespacedElements(doc.documentElement, unwanted_ns) > 0:
-            pass
+        _num_elements_removed += removeNamespacedElements(doc.documentElement,
```

Comment:
Would be great if this were true, but someone added the comment above for some reason at some point, so I wonder if the loop can really be removed?

Reply:
Honestly, I cannot see how that comment can still be true.  Unfortunately, both the functions and the comments predates the version history in git, so it is not possible to tell if it underlying problem was fixed and the comments were just left there to rot.

But if the comments are true, then we are missing a regression test case to show the issue.  Sadly, it is a bit of a devil's proof and my head is convinced that there is no such case (doesn't make it true; I cannot find a case but my thinking is biased).

I am fine with ripping those two commits out of the branch if you are not comfortable with them.

Reply:
I think you're right, let's keep the commits and remove the comment!

Note you can [get full history](https://github.com/scour-project/scour/issues/23#issuecomment-241896530) with `git fetch origin 'refs/replace/*:refs/replace/*'` or via [old bzr history](https://bazaar.launchpad.net/~scouring/scour/trunk/changes) on Launchpad.

Looking at that, the comment was added in https://github.com/scour-project/scour/commit/879300373f39319c76ef58bd10ce0e32ee55d22b and fixed already in https://github.com/scour-project/scour/commit/2dc788aa3f5196a14c00262238ba838248a48077.